packages/account/src/predicate/predicate.ts

```diff
@@ -255,4 +291,37 @@ export class Predicate<TInputData extends InputValue[]> extends Account {
 
     return mutatedBytes;
   }
+
+  private getIndexFromPlaceholderWitness(request: TransactionRequest): number {
```

Comment:
Not really sure how the index would be negative? Maybe a short doc on this function would be beneficial.

Reply:
@danielbate The index is negative when removing an entry from the `TransactionRequest.witnesses` is unnecessary. I will add a description

Reply:
https://github.com/FuelLabs/fuels-ts/pull/2037/commits/427dbd655c5d066955d5142e3965e6b96b135036